tests/main/core-backlight/task.yaml

```
@@ -0,0 +1,22 @@
+summary: Test system.disable-backlight-service core config option
+
+systems: [ubuntu-core-*]
+
+environment:
+    MASKFILE: /etc/systemd/system/systemd-backlight@.service
+
+execute: |
+    echo "Backlight is not masked by default"
+    systemctl status systemd-backlight@foo | MATCH "Loaded: loaded"
+
+    echo "Check that backlight service can be disabled"
+    snap set core system.disable-backlight-service=true
+
+    systemctl status systemd-backlight@foo | MATCH "Loaded: masked"
+    test -L "$MASKFILE"
+        
+    echo "Check that backlight service can be enabled"
+    snap set core system.disable-backlight-service=false
+
+    systemctl status systemd-backlight@foo | MATCH "Loaded: loaded"
+    ! test -L "$MASKFILE"
```

I think you want to test that the file is simply not present anymore, right?
Also gotcha about ! foo being useless for set -e.

```
zyga@fyke:~$ PS1='foo '
foo set -e
foo false  # bash quits because -e
zyga@fyke:~$ bash
zyga@fyke:~$ PS1='foo '
foo set -e
foo ! true
foo  # surprise, this is not what you expected.
```

```suggestion
    not test -e "$MASKFILE"
```